File: python/cnote.py
UNLUCKY = "UnluckyChef"
LUCKY = "LuckyChef"

T = int(input())
chefLuck = []

def decideLuck(pages_left, pages_needed, budget, books_in_shop):
    chef = UNLUCKY
    if pages_needed <= pages_left:
        chef = LUCKY
    for i in range(books_in_shop):
        number_of_pages, cost = map(int, input().split())
        if pages_needed - pages_left <= number_of_pages and cost <= budget:
            chef = LUCKY
    return chef

for i in range(T):
    pages_needed, pages_left, budget, books_in_shop = map(int, input().split())
    chefLuck.append(decideLuck(pages_left, pages_needed, budget, books_in_shop))
for value in chefLuck:
    print(value)

File: java/CNote.java
import java.util.Scanner;
class CNote {
	int pagesLeft;
	int pagesNeeded;
	int budget;
	int booksCount;
	public final String LUCKY = "LuckyChef";
	public final String UNLUCKY = "UnluckyChef";
	public CNote() {
	}
	public void setValues( int pagesLeft,
		int pagesNeeded,
		int budget,
		int booksCount ) {
		this.pagesLeft = pagesLeft;
		this.pagesNeeded = pagesNeeded;
		this.budget = budget;
		this.booksCount = booksCount;
	}
	public String decideLuck(int numberOfPages[], int cost[]) {
		String lucky = UNLUCKY;
		if(pagesNeeded <= pagesLeft)	lucky = LUCKY;
		for(int i = 0; i < booksCount; i++) {
			if(pagesNeeded - pagesLeft <= numberOfPages[i])
				if(cost[i] <= budget)
					lucky = LUCKY;
		}
		return lucky;
	}
	public static void main(String args[]) {
		Scanner s = new Scanner(System.in);
		int T = s.nextInt();
		CNote cnote = new CNote();
		String[] chefLuck = new String[T];
		for(int i = 0; i < T; i++) {
			int pagesNeeded = s.nextInt();
			int pagesLeft = s.nextInt();
			int budget = s.nextInt();
			int booksCount = s.nextInt();
			cnote.setValues(pagesLeft, pagesNeeded, budget, booksCount);
			int[] numberOfPages = new int[booksCount];
			int[] cost = new int[booksCount];	
			for(int j = 0; j < booksCount; j++) {
				numberOfPages[j] = s.nextInt();
				cost[j] = s.nextInt();
			}
			chefLuck[i] = cnote.decideLuck(numberOfPages, cost);
		}
		for(int i = 0; i < T; i++) {
			System.out.println(chefLuck[i] + " ");
		}
	}
}
